tests/framework/builder.py

```diff
@@ -202,24 +220,44 @@ def create(self,
                target_version: str = None,
                mem_file_name: str = "vm.mem",
                snapshot_name: str = "vm.vmstate",
-               net_ifaces=None):
+               net_ifaces=None,
+               use_ramdisk=False):
         """Create a Snapshot object from a microvm and artifacts."""
+        if use_ramdisk:
+            snaps_dir = self._microvm.jailer.chroot_ramfs_path()
+            mem_full_path = os.path.join(snaps_dir, mem_file_name)
+            vmstate_full_path = os.path.join(snaps_dir, snapshot_name)
+
+            # Pre-allocate ram for memfile to eliminate allocation variability.
+            utils.run_cmd(
+                'dd if=/dev/zero of={} bs=1M count=512'.format(mem_full_path)
```

Comment:
Why is the memfile size hardcoded here? It could change in the future

Reply:
If this is just the preallocation, then the ramdisk can grow more than this, right?

Reply:
I'm referriing to the size of the memfile, which here is allocated to 512M.
There are also microvms in the snapshot perf test that have a memory of 256M, and in the future we may add VMs with more memory maybe?

I'm saying that this should be configurable